src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs

```diff
@@ -189,32 +200,14 @@ private static void ReadCallback(UvStreamHandle handle, int status, object state
             }
         }
 
-        private void Pause()
+        private void StopReading()
         {
-            // It's possible that uv_close was called between the call to Thread.Post() and now.
-            if (!_socket.IsClosed)
-            {
-                _socket.ReadStop();
-            }
+            _socket.ReadStop();
         }
 
-        private void Resume()
+        private void StartReading()
         {
-            // It's possible that uv_close was called even before the call to Resume().
-            if (!_socket.IsClosed)
-            {
-                try
-                {
-                    _socket.ReadStart(_allocCallback, _readCallback, this);
-                }
-                catch (UvException)
-                {
-                    // ReadStart() can throw a UvException in some cases (e.g. socket is no longer connected).
-                    // This should be treated the same as OnRead() seeing a "normalDone" condition.
-                    Log.ConnectionReadFin(ConnectionId);
-                    Input.Complete();
-                }
-            }
+            _socket.ReadStart(_allocCallback, _readCallback, this);
```

Comment:
I can add the try catch back but I'd like to understand if it's truly needed (and write a test for it maybe).